### ISSUE TITLE
Pipeline support

### ISSUE THREAD

**Issue by pupssman** [jenkinsci/allure-plugin#49]
Unfortunately, one cannot use allure report generation in any way from the Pipeline scripts (https://jenkins.io/doc/pipeline/)

Lets contribute a `SimpleBuildStep` implementation that builds allure report!

See https://github.com/jenkinsci/pipeline-plugin/blob/master/DEVGUIDE.md

**Comment by pazone** [comment]
@pupssman did you tried common HTML publisher?

**Comment by etki** [comment]
would love to see this as a standalone feature as well

**Comment by dhanish2k** [comment]
I think this would be great feature.

**Comment by arielcloud** [comment]
+1

**Comment by Narkhip** [comment]
+1

**Comment by ludenus** [comment]
Just in case.
I have successfully published Allure report (created by maven plugin) via HTML Publisher plugin

```
    stage('Publish') {
        echo 'Publish Allure report'
        archive 'target/*.jar'
        publishHTML(
                target: [
                        allowMissing         : false,
                        alwaysLinkToLastBuild: false,
                        keepAll              : true,
                        reportDir            : 'target/site/allure-maven-plugin',
                        reportFiles          : '*index*',
                        reportName           : "Allure Report"
                ]
        )
    }
```

details here: http://stackoverflow.com/questions/40379494/how-does-one-run-allure-plugin-in-jenkins-pipeline/40911564#40911564

**Comment by buzdin** [comment]
Don't forget to put an example in confluence when pipelines are supported. And +1 for this feature.

**Comment by pupssman** [comment]
@eroshenkoam @baev wasn't this fixed in #70?

**Comment by etki** [comment]
@pupssman If i'm right, this hasn't been officially released to upstream yet

**Comment by eroshenkoam** [comment]
fixed via #70
